types/lib/model.d.ts

```diff
@@ -2300,7 +2306,10 @@ export abstract class Model<T = any, T2 = any> extends Hooks {
    * @param target The model that will be associated with hasOne relationship
    * @param options Options for the association
    */
-  public static hasMany(target: typeof Model, options?: HasManyOptions): HasMany;
+  public static hasMany<M, T extends typeof Model>(
+    this: M, 
+    target: T, options?: HasManyOptions
```

Comment:
Nit: Please be consistent about newlines

Reply:
done.

It could be nice to have some prettier config